t/unit/test_connection.py

```diff
@@ -1045,6 +1045,31 @@ def test_acquire_channel(self):
         with P.acquire_channel() as (conn, channel):
             assert channel is conn.default_channel
 
+    def test_exception_during_connection_use(self):
```

Comment:
Very cool test!!
Please document what it does.

It will be usable in other use-cases, so it's better this specific test has proper documentation.

Nice job.

P.s
We don't have many good tests for connection loss edge cases (in general), so this instance will be educational 🚀

Reply:
Thanks for the very quick feedback. I changed the test slightly (mocking replace is not needed).
Instead I am now checking that the connection is closed in case of an exception.